AutoCollection/Statsbeat.ts

```diff
@@ -0,0 +1,263 @@
+import os = require("os");
+import EnvelopeFactory = require("../Library/EnvelopeFactory");
+import Logging = require("../Library/Logging");
+import Sender = require("../Library/Sender");
+import Constants = require("../Declarations/Constants");
+import Contracts = require("../Declarations/Contracts");
+import Vm = require("../Library/AzureVirtualMachine");
+import Config = require("../Library/Config");
+import Context = require("../Library/Context");
+
+const STATSBEAT_LANGUAGE = "node";
+const STATSBEAT_CONNECTIONSTRING = "InstrumentationKey=c4a29126-a7cb-47e5-b348-11414998b11e;IngestionEndpoint=https://dc.services.visualstudio.com/";
+
+
+class Statsbeat {
+
+    private static TAG = "Statsbeat";
+
+    // Counters
+    private _lastRequests: { totalRequestCount: number; time: number; };
+    private _totalRequestCount: number = 0;
+    private _totalSuccesfulRequestCount: number = 0;
+    private _totalFailedRequestCount: number = 0;
+    private _retryCount: number = 0;
+    private _exceptionCount: number = 0;
+    private _throttleCount: number = 0;
+    private _intervalRequestExecutionTime: number = 0;
+    private _lastIntervalRequestExecutionTime: number = 0;
+
+    private _collectionInterval: number = 900000; // 15 minutes
+    private _sender: Sender;
+    private _handle: NodeJS.Timer | null;
+    private _isEnabled: boolean;
+    private _isInitialized: boolean;
+    private _config: Config;
+    private _statsbeatConfig: Config;
+    private _isVM: boolean;
+    private _statbeatMetrics: Array<{ name: string; value: number }>;
+
+    // Custom dimensions
+    private _resourceProvider: string;
+    private _sdkVersion: string;
+    private _runtimeVersion: string;
+    private _os: string;
+    private _language: string;
+    private _cikey: string;
+    private _attach: string = Constants.StatsbeatAttach.sdk; // Default is SDK
+    private _features: number = Constants.StatsbeatFeature.NONE;
+    private _instrumentations: number = Constants.StatsbeatInstrumentation.NONE;
+
+    constructor(config?: Config) {
+        this._isInitialized = false;
+        this._statbeatMetrics = [];
+        this._config = config;
+        this._statsbeatConfig = new Config(STATSBEAT_CONNECTIONSTRING);
```

Comment:
it would be nice if you can get statsbeatConnectionString from the input parameter Config, then you can add statsbeat test to the e2e testing suite :)

Reply:
maybe add a todo for later if you like.

Reply:
I created a task for that, current there is no way to configure in attach scenarios but it would be great to have something available